src/Plugin.php

```diff
@@ -64,6 +64,10 @@ public static function product_rewrite_rules(array $rules) {
    */
   public static function request(array $query_vars) {
     if (isset($query_vars['product_cat']) && $query_vars['product_cat'] !== '' && !term_exists($query_vars['product_cat'], 'product_cat')) {
+      $pagename = static::getCategoryBase() . '/' . ltrim($query_vars['product_cat_and_post_name'], '/');
```

Comment:
Last remaining step is to add a comment here to explain what is being done and why.